### PR TITLE
5 3 manage beneficiaries cypress tests

### PR DESCRIPTION
**Pull request by DurkoMatko**
add: manage beneficiaries tests (delete, merge, detach, deactivate, reactivate)
change: tiny changes in add beneficiary tests

@james I can't make these work without using cy.wait. It's after beneficiary creating/merging/detaching operations when I try to check their checkbox. Cypress doesn't wait till the page with the list loads...do you maybe have some idea?

### CODE REVIEW

**Comment by jamescrowley** [comment]
Quick update. So, these waits() are definitely due to the fact we're doing:

- Ajax POST to perform an action, which returns a json object, with a redirect: true flag
- then a client-side redirect is triggered

and cypress is only waiting for the first to complete, it has no idea when the page has re-loaded or not or whether it even will (https://github.com/cypress-io/cypress/issues/1805)

I'm not sure how to fix this yet - it's a pretty weird flow in the first place.

File: cypress/integration/5_3_Manage_Beneficiaries.js [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        fullDeleteTestedBeneficiaries([TEST_FIRSTNAME1,TEST_FIRSTNAME2]);   //delete beneficiaries from previous tests (should not be any, but just in case)
+        createMergedFamily(TEST_FIRSTNAME1, TEST_LASTNAME1, TEST_FIRSTNAME2, TEST_LASTNAME2, TEST_CASE_ID);
+        cy.wait(7000);
+        checkBeneficiaryCheckboxByName(TEST_LASTNAME2);
+        clickDetachButton();
+        cy.wait(7000);
+        verifyBeneficiaryRowLevel(TEST_LASTNAME1,0);
+        verifyBeneficiaryRowLevel(TEST_LASTNAME2,0);
+        //cleanup
+        fullDeleteTestedBeneficiaries([TEST_FIRSTNAME1,TEST_FIRSTNAME2]);
```
Comment (naphets123) [inline cypress/integration/5_3_Manage_Beneficiaries.js]: As above Delete in beginning and end?

File: cypress/integration/5_3_Manage_Beneficiaries.js [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+        cy.wait(7000);
+        checkBeneficiaryCheckboxByName(TEST_LASTNAME1);
+        checkBeneficiaryCheckboxByName(TEST_LASTNAME2);
+        clickMergeButton();
+        cy.wait(7000);
+        verifyBeneficiaryRowLevel(TEST_LASTNAME1,0);
+        verifyBeneficiaryRowLevel(TEST_LASTNAME2,1);
+        
+        //cleanup
+        fullDeleteTestedBeneficiaries([TEST_FIRSTNAME1,TEST_FIRSTNAME2]);
```
[the remark this answers is not in the document]
Comment (DurkoMatko) [reply cypress/integration/5_3_Manage_Beneficiaries.js]: James pointed this out in another PR of mine. The reason is that I'm simply not confident that the cleanup will be successful (cuz of the flakyness of the tests - which actually isn't that bad, but in general, I don't like the approach we have -> clicking the DOM). But yeah, you're right James pointed it out as well. Maybe I should trust cypress more and be happy with just one try :D

File: cypress/integration/5_3_Manage_Beneficiaries.js [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+    it('Merge beneficiaries into family', () => {
+        fullDeleteTestedBeneficiaries([TEST_FIRSTNAME1,TEST_FIRSTNAME2]);   //delete beneficiaries from previous tests (should not be any, but just in case)
+        createTestBeneficiary(TEST_FIRSTNAME1, TEST_LASTNAME1, TEST_CASE_ID);
+        createTestBeneficiary(TEST_FIRSTNAME2, TEST_LASTNAME2, TEST_CASE_ID);
+        cy.wait(7000);
+        checkBeneficiaryCheckboxByName(TEST_LASTNAME1);
+        checkBeneficiaryCheckboxByName(TEST_LASTNAME2);
+        clickMergeButton();
+        cy.wait(7000);
```
[the remark this answers is not in the document]
Comment (DurkoMatko) [reply cypress/integration/5_3_Manage_Beneficiaries.js]: Have you tried with different wait time? 7000 is just a random number which worked for me in my PC :D

File: cypress/integration/5_3_Manage_Beneficiaries.js [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+        getDeactivatedTab().should('have.class', 'active');
+        getBeneficiaryRow(DEACTIVATED_BENEFICIARY).should('exist');
+    });
+
+    //DOESNT WORK YET!
+    it('Recover deactivated beneficiaries', () => {
+        //make sure there isn't a beneficiary with the same name in the All tab
+        cy.get('body').then(($body) => {
+            if ($body.text().includes(TEST_LASTNAME3)) {
+                cy.log("found" + TEST_LASTNAME3)
```
[the remark this answers is not in the document]
Comment (DurkoMatko) [reply cypress/integration/5_3_Manage_Beneficiaries.js]: Hmm I think the assertion `.should('be.visible')` throws false if what you're looking for is not visible, or? The function deletes user if he exists, but if he doesn't it shouldn't fail the test. 

Im not sure what you mean by `cy.log` tho...it's just a log message, it doesn't have any functionality. I've actually stolen it from your conditional boxes cleanup function to make the conditional cleanup logging consistent. But we can for sure just delete it from both functions as it's only place where we use cypress logging anyway.

**Comment by HaGuesto** [comment]
@DurkoMatko please resolve the conflict and try to replace wait with reload.

File: cypress/integration/5_3_Manage_Beneficiaries.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (DurkoMatko) [reply cypress/integration/5_3_Manage_Beneficiaries.js]: Ok, these were replaces by cy.reload now.

File: cypress/integration/5_3_Manage_Beneficiaries.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (DurkoMatko) [reply cypress/integration/5_3_Manage_Beneficiaries.js]: Commented out the second cleanup.

File: cypress/integration/5_3_Manage_Beneficiaries.js (hunk 1 above)
Comment (DurkoMatko) [reply cypress/integration/5_3_Manage_Beneficiaries.js]: Commented out the one in the beginning.

File: cypress/integration/5_3_Manage_Beneficiaries.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (DurkoMatko) [reply cypress/integration/5_3_Manage_Beneficiaries.js]: Dunno what to do with this one. I guess it's just a log and can stay in there, or?

**Comment by DurkoMatko** [comment]
@jamescrowley: As adviced by @HaGuesto and @naphets123 who already faced similar issue, I've replaced waits with cy.reload. So this should be ready to be reviewed again and hopefully merged.

**Pull request merged**